package/health_daemon/src/baseline_thread.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2017 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <float.h>
+#include <math.h>
+#include <stdlib.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/sbp_zmq_pubsub.h>
+
+#include <libsbp/sbp.h>
+#include <libsbp/navigation.h>
+
+#include "health_thread.h"
+
+#include "baseline_thread.h"
+
+#define BASELINE_THRESHOLD (80000.0f)
+static health_thread_t* baseline_thread;
+
+static int sbp_msg_baseline_ecef_callback(health_thread_t* thread, u16 sender_id, u8 len, u8 msg_[], void *ctx)
+{
+  (void)thread;
+  (void)sender_id;
+  (void)len;
+  log_fn_t log_fn = (log_fn_t)ctx;
+  msg_baseline_ecef_t *msg = (void*)msg_;
+  float x = msg->x, y = msg->y, z = msg->z;
+  float distance = sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2));
+  if (distance > BASELINE_THRESHOLD) {
+    log_fn(LOG_WARNING, "Baseline Distance Over Threshold: %fm", distance);
```

Comment:
How often is this warning going to be issued?  We may want to rate limit to every 5 or 10s?